notion_client/helpers.py

```diff
@@ -124,3 +134,169 @@ def is_equation_rich_text_item_response(rich_text: Dict[Any, Any]) -> bool:
+            if fragment.startswith("block-"):
+                fragment = fragment[6:]
+            if compact_uuid_pattern.match(fragment):
+                return _format_uuid(fragment.lower())
+
+        return None
+
+    except Exception:
+        return None
+
+
+def extract_database_id(url_or_id: str) -> Optional[str]:
+    """Extract a database ID from a Notion URL or validate if it's already a valid ID.
+
+    This is an alias for extract_notion_id for clarity when working with databases.
+
+    Args:
+        url_or_id: A Notion database URL or ID string
+
+    Returns:
+        The extracted UUID in standard format (with hyphens) or None if invalid
+    """
+    return extract_notion_id(url_or_id)
+
+
+def extract_page_id(url_or_id: str) -> Optional[str]:
+    """Extract a page ID from a Notion URL or validate if it's already a valid ID.
+
+    This is an alias for extract_notion_id for clarity when working with pages.
+
+    Args:
+        url_or_id: A Notion page URL or ID string
+
+    Returns:
+        The extracted UUID in standard format (with hyphens) or None if invalid
+    """
+    return extract_notion_id(url_or_id)
+
+
+def extract_block_id(url_or_id: str) -> Optional[str]:
+    """Extract a block ID from a Notion URL fragment or validate if it's already a valid ID.
+
+    Specifically looks for block IDs in URL fragments (after #).
+    If no fragment is present, falls back to extract_notion_id behavior.
+
+    Args:
+        url_or_id: A Notion URL with block fragment or ID string
+
+    Returns:
+        The extracted UUID in standard format (with hyphens) or None if invalid
+    """
+    if not url_or_id or not isinstance(url_or_id, str):
+        return None
+
+    # If it looks like a URL, try to extract from fragment first
+    if "://" in url_or_id:
+        try:
+            parsed = urlparse(url_or_id.strip())
+            if parsed.fragment:
+                fragment = parsed.fragment
+                # Remove "block-" prefix if present
+                if fragment.startswith("block-"):
+                    fragment = fragment[6:]
+
+                # Check if it's a compact UUID
+                compact_uuid_pattern = re.compile(r"^[0-9a-f]{32}$", re.IGNORECASE)
+                if compact_uuid_pattern.match(fragment):
+                    return _format_uuid(fragment.lower())
+
+                # Check if it's already formatted UUID
+                uuid_pattern = re.compile(
+                    r"^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$",
+                    re.IGNORECASE,
+                )
+                if uuid_pattern.match(fragment):
+                    return fragment.lower()
+
+            # If no valid fragment found, return None (block IDs should be in fragments)
+            return None
+        except Exception:
+            pass
+
+    # Fall back to general ID extraction for non-URL inputs
+    return extract_notion_id(url_or_id)
```

Comment:
Why is the implementation different than in notion-sdk-js? https://github.com/makenotion/notion-sdk-js/blob/6945289f774aa0150d9c382b1f88c752dbe0f925/src/helpers.ts#L304-L308